test/onnx/gen_onnx.py

```diff
@@ -2009,6 +2009,751 @@ def dynamicquantizelinear_2d_test():
+                                 outputs=['y'],
+                                 equation='ij->')
+
+    return ([node], [x], [y])
+
+
+@onnx_test()
+def einsum_column_sum_test():
+    x = helper.make_tensor_value_info('x', TensorProto.FLOAT, [2, 3])
+    y = helper.make_tensor_value_info('y', TensorProto.FLOAT, [3])
+
+    node = onnx.helper.make_node('Einsum',
+                                 inputs=['x'],
+                                 outputs=['y'],
+                                 equation='ij->j')
+
+    return ([node], [x], [y])
+
+
+@onnx_test()
+def einsum_row_sum_test():
+    x = helper.make_tensor_value_info('x', TensorProto.FLOAT, [2, 3])
+    y = helper.make_tensor_value_info('y', TensorProto.FLOAT, [2])
+
+    node = onnx.helper.make_node('Einsum',
+                                 inputs=['x'],
+                                 outputs=['y'],
+                                 equation='ij->i')
+
+    return ([node], [x], [y])
+
+
+@onnx_test()
+def einsum_matrix_vector_multiplication_test():
+    x = helper.make_tensor_value_info('x', TensorProto.FLOAT, [2, 3])
+    v = helper.make_tensor_value_info('v', TensorProto.FLOAT, [1, 3])
+    y = helper.make_tensor_value_info('y', TensorProto.FLOAT, [2, 1])
+
+    node = onnx.helper.make_node('Einsum',
+                                 inputs=['x', 'v'],
+                                 outputs=['y'],
+                                 equation='ij,kj->ik')
+
+    return ([node], [x, v], [y])
+
+
+@onnx_test()
+def einsum_matrix_matrix_multiplication_test():
+    x1 = helper.make_tensor_value_info('x1', TensorProto.FLOAT, [2, 3])
+    x2 = helper.make_tensor_value_info('x2', TensorProto.FLOAT, [2, 3])
+    y = helper.make_tensor_value_info('y', TensorProto.FLOAT, [2, 2])
+
+    node = onnx.helper.make_node('Einsum',
+                                 inputs=['x1', 'x2'],
+                                 outputs=['y'],
+                                 equation='ij,kj->ik')
+
+    return ([node], [x1, x2], [y])
+
+
+@onnx_test()
+def einsum_vector_dot_product_test():
+    x1 = helper.make_tensor_value_info('x1', TensorProto.FLOAT, [3])
+    x2 = helper.make_tensor_value_info('x2', TensorProto.FLOAT, [3])
+    y = helper.make_tensor_value_info('y', TensorProto.FLOAT, [1])
+
+    node = onnx.helper.make_node('Einsum',
+                                 inputs=['x1', 'x2'],
+                                 outputs=['y'],
+                                 equation='i,i->')
+
+    return ([node], [x1, x2], [y])
+
+
+@onnx_test()
+def einsum_matrix_dot_product_test():
+    x1 = helper.make_tensor_value_info('x1', TensorProto.FLOAT, [2, 3])
+    x2 = helper.make_tensor_value_info('x2', TensorProto.FLOAT, [2, 3])
+    y = helper.make_tensor_value_info('y', TensorProto.FLOAT, [1])
+
+    node = onnx.helper.make_node('Einsum',
+                                 inputs=['x1', 'x2'],
+                                 outputs=['y'],
+                                 equation='ij,ij->')
```

Comment:
Do we have tests without the `->` to make sure that works?

Reply:
Yes, we have a few:
- https://github.com/ROCm/AMDMIGraphX/blob/3dceee176d4beb3e5877f85a9772ef6ffca2ec93/test/onnx/gen_onnx.py#L2258
- https://github.com/ROCm/AMDMIGraphX/blob/3dceee176d4beb3e5877f85a9772ef6ffca2ec93/test/onnx/gen_onnx.py#L2271
- https://github.com/ROCm/AMDMIGraphX/blob/3dceee176d4beb3e5877f85a9772ef6ffca2ec93/test/onnx/gen_onnx.py#L2413
- https://github.com/ROCm/AMDMIGraphX/blob/3dceee176d4beb3e5877f85a9772ef6ffca2ec93/test/onnx/gen_onnx.py#L2427